examples/DataApi-NFT/scripts/send_data_transaction.js

```diff
@@ -0,0 +1,119 @@
+const { dag4 } = require("@stardust-collective/dag4");
+const jsSha256 = require("js-sha256");
+const axios = require("axios");
+
+const buildMintCollection = () => {
+  return {
+    MintCollection: {
+      name: ":collectionName",
+    },
+  };
+};
+
+const buildMintNFT = () => {
+  return {
+    MintNFT: {
+      owner: ":ownerAddress",
+      collectionId: ":collectionId",
+      nftId: ":nftIdAsNumber",
+      uri: ":validNFTURL",
+      name: ":nftName",
+      description: ":nftDescription",
+      metadata: {
+        key: "value",
+      },
+    },
+  };
+};
+
+const buildTransferCollection = () => {
+  return {
+    TransferCollection: {
+      fromAddress: ":fromAddress",
+      toAddress: ":toAddress",
+      collectionId: ":collectionId",
+    },
+  };
+};
+
+const buildTransferNFT = () => {
+  return {
+    TransferCollection: {
+      fromAddress: ":fromAddress",
+      toAddress: ":toAddress",
+      collectionId: ":collectionId",
+      nftId: ":nftIdAsNumber",
+    },
+  };
+};
+
+/** Encode message according with serializeUpdate on your template module l1 */
+const getEncoded = (value) => {
+  const energyValue = JSON.stringify(value);
+  return energyValue;
+};
+
+const serialize = (msg) => {
+  const coded = Buffer.from(msg, "utf8").toString("hex");
+  return coded;
+};
+
+const generateProof = async (message, walletPrivateKey, account) => {
+  const encoded = getEncoded(message);
+  console.log(encoded);
+
+  const serializedTx = serialize(encoded);
+  const hash = jsSha256.sha256(Buffer.from(serializedTx, "hex"));
+  const signature = await dag4.keyStore.sign(walletPrivateKey, hash);
+
+  const publicKey = account.publicKey;
+  const uncompressedPublicKey =
+    publicKey.length === 128 ? "04" + publicKey : publicKey;
+
+  return {
+    id: uncompressedPublicKey.substring(2),
+    signature,
+  };
+};
+
+const sendDataTransactionsUsingUrls = async (
+  globalL0Url,
+  metagraphL1DataUrl
+) => {
+  const walletPrivateKey = ":private_key";
+
+  const account = dag4.createAccount();
+  account.loginPrivateKey(walletPrivateKey);
+
+  account.connect({
+    networkVersion: "2.0",
+    l0Url: globalL0Url,
+    testnet: true,
+  });
+
+  const message = buildMintCollection();
+  const proof = await generateProof(message, walletPrivateKey, account);
```

Comment:
We should add this to dag4 so that it doesn't need to be repeated all over

Reply:
Yeah, we can think in add this